### PR TITLE
Add all values to entrypoint

### PR DESCRIPTION
**Pull request by munishchouhan**
This PR will change the current functionality of creating entrypoint, instead of only using the first value of the input list, it will join all the values in list

### CODE REVIEW

**Comment by munishchouhan** [comment]
@jordeu please check these tests, if these are the correct use cases

File: src/main/groovy/io/seqera/wave/core/ContainerAugmenter.groovy [hunk 1]
```diff
@@ -303,12 +305,13 @@ class ContainerAugmenter {
         return new Tuple2<String, Integer>(digest, size)
     }
 
-    protected String getFirst(value) {
+    protected static String getFirst(value) {
```
Comment (jordeu) [inline src/main/groovy/io/seqera/wave/core/ContainerAugmenter.groovy:308]: Maybe we can rename `getFirst` to something more meaningful

File: src/test/groovy/io/seqera/wave/core/ContainerAugmenterTest.groovy [hunk 2]
```diff
@@ -856,4 +856,17 @@ class ContainerAugmenterTest extends Specification {
...
+        ContainerAugmenter.getFirst(INPUT_VALUE) == ENTRYPOINT
+
+        where:
+        INPUT_VALUE                         | ENTRYPOINT
+        null                                |null
+        'entrypoint'                        |'entrypoint'
+        ['entrypoint', 'list']              |'["entrypoint","list"]'
+        ["/bin/sh", "-c", "\"echo Hola\""]  |'["/bin/sh","-c","\\"echo Hola\\""]'
+        ["foo", "bar foo"]                  |'["foo","bar foo"]'
+        ["foo", "'bar foo'"]                |'["foo","\'bar foo\'"]'
```
Comment (jordeu) [inline src/test/groovy/io/seqera/wave/core/ContainerAugmenterTest.groovy:870]: Add a single-item list case to check backward compatibility. Ex: `["entrypoint"] --> "entrypoint"`

File: src/main/groovy/io/seqera/wave/core/ContainerAugmenter.groovy (hunk 1 above)
Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/core/ContainerAugmenter.groovy]: done

File: src/test/groovy/io/seqera/wave/core/ContainerAugmenterTest.groovy (hunk 2 above)
Comment (munishchouhan) [reply src/test/groovy/io/seqera/wave/core/ContainerAugmenterTest.groovy:871]: done

**Review by jordeu — APPROVED** [review]

File: src/main/groovy/io/seqera/wave/core/ContainerAugmenter.groovy [hunk 3]
```diff
@@ -34,6 +34,8 @@ import io.seqera.wave.core.spec.ManifestSpec
 import io.seqera.wave.exception.DockerRegistryException
 import io.seqera.wave.proxy.ProxyClient
 import io.seqera.wave.storage.Storage
+import io.seqera.wave.util.Escape
```
Comment (pditommaso) [inline src/main/groovy/io/seqera/wave/core/ContainerAugmenter.groovy:37]:
```suggestion
```

File: src/test/groovy/io/seqera/wave/core/ContainerAugmenterTest.groovy [hunk 4]
```diff
@@ -856,4 +856,18 @@ class ContainerAugmenterTest extends Specification {
...
+
+        where:
+        INPUT_VALUE                         | ENTRYPOINT
+        null                                |null
+        'entrypoint'                        |'entrypoint'
+        ['entrypoint']                      |'entrypoint'
+        ['entrypoint', 'list']              |'["entrypoint","list"]'
+        ["/bin/sh", "-c", "\"echo Hola\""]  |'["/bin/sh","-c","\\"echo Hola\\""]'
+        ["foo", "bar foo"]                  |'["foo","bar foo"]'
+        ["foo", "'bar foo'"]                |'["foo","\'bar foo\'"]'
```
Comment (pditommaso) [inline src/test/groovy/io/seqera/wave/core/ContainerAugmenterTest.groovy:871]:
```suggestion
        null                                | null
        'entrypoint'                        | 'entrypoint'
        ['entrypoint']                      | 'entrypoint'
        ['entrypoint', 'list']              | '["entrypoint","list"]'
        ["/bin/sh", "-c", "\"echo Hola\""]  | '["/bin/sh","-c","\\"echo Hola\\""]'
        ["foo", "bar foo"]                  | '["foo","bar foo"]'
        ["foo", "'bar foo'"]                | '["foo","\'bar foo\'"]'
```

**Review by pditommaso — APPROVED** [review]

**Pull request merged**